include/core/simulation_control.h

```diff
@@ -697,4 +719,43 @@ class SimulationControlAdjointSteady : public SimulationControlTransient
   is_at_end() override;
 };
 
+class SimulationControlRayTracing : public SimulationControl
+{
+public:
+  SimulationControlRayTracing(
+    const Parameters::SimulationControl &param,
+    Particles::ParticleHandler<3>       &input_photon_handler);
```

Comment:
You are missing brief and documentation for your constructor and your integrate function :) .